### PR TITLE
ASoC: SOF: core: Print out the value of sof_debug if it is set

### PR DESCRIPTION
**Pull request by ujfalusi**
The sof_debug value is set by the user, developer intentionally. To save time on figuring out what value has been passed to the kernel by the user, developer, print it out if it is not 0.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

### CODE REVIEW

File: sound/soc/sof/core.c
Comment (dbaluta) [inline sound/soc/sof/core.c:367]: I think it would make sense to print it every time and use dev_dbg instead of dev_info.

Comment (plbossart) [reply sound/soc/sof/core.c:367]: it's true that it'll be printed every time for Intel devices since BIT(0) is always set.

Comment (ujfalusi) [reply sound/soc/sof/core.c:367]: @plbossart, not in distro kernels, I think the CONFIG_SND_SOC_SOF_DEBUG_ENABLE_FIRMWARE_TRACE is not set by default.

Comment (plbossart) [reply sound/soc/sof/core.c:367]: i am ok with dev_info.

Comment (ujfalusi) [reply sound/soc/sof/core.c:367]: OK, should I re-send with some change, or?

**Review by plbossart — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
This may be used to set  functional aspects (like disabling multicore), so it is possible sof_debug is used in scenarios where dynamic debug is disabled (e.g. chrome kernels). So dev_info() seems good to me.

**Review by dbaluta — APPROVED** [review]

File: sound/soc/sof/core.c
Comment (dbaluta) [reply sound/soc/sof/core.c:367]: @ujfalusi lets merge it like that. I think the `if(sof_core_debug)` will avoid spamming production kernels.

Comment (ujfalusi) [reply sound/soc/sof/core.c:367]: One thing came to mind before merging this: Do we have production kernel(s) where the NOCODEC mode is the default, which from now on can be selected via a sof_debug flag?
That would be not so nice in the logs.

Comment (plbossart) [reply sound/soc/sof/core.c:367]: I don't see how NOCODEC could be used in production kernels. That is clearly spelled out in the Kconfig

````
config SND_SOC_SOF_NOCODEC_SUPPORT
	bool "SOF nocodec static mode support"
	help
	  This adds support for a dummy/nocodec machine driver fallback
	  option if no known codec is detected. This is typically only
	  enabled for developers or devices where the sound card is
	  controlled externally.
	  This option is mutually exclusive at build time with the Intel HDAudio support.
	  Selecting it may have negative impacts and prevent e.g. microphone
	  functionality from being enabled on Intel CoffeeLake and later
	  platforms.
	  Distributions should not select this option!
````

**Comment by ranj063** [comment]
@ujfalusi is this really necessary. A user can always read it from sysfs no? For example:

root@ranjani-Voxel:/sys/module/snd_sof_pci/parameters# cat sof_pci_debug
1024

**Comment by ujfalusi** [comment]
> @ujfalusi is this really necessary. A user can always read it from sysfs no? For example:
> 
> root@ranjani-Voxel:/sys/module/snd_sof_pci/parameters# cat sof_pci_debug 1024

Right, but if you receive the dmesg only (like a log from CI or other automation system) then you will have hard time to access to the sysfs file (and I had no idea where the parameter is in there before you pointed it out now).

hrm:
```
# dmesg | grep sof_debug
[18489.604302] sof-audio-pci-intel-tgl 0000:00:1f.3: sof_debug value: 0x8201
```
```
# cat /sys/module/snd_sof_pci/parameters/sof_pci_debug
0
```
something is not adding up?

**Comment by ujfalusi** [comment]
> ```
> # dmesg | grep sof_debug
> [18489.604302] sof-audio-pci-intel-tgl 0000:00:1f.3: sof_debug value: 0x8201
> ```
> 
> ```
> # cat /sys/module/snd_sof_pci/parameters/sof_pci_debug
> 0
> ```
> 
> something is not adding up?

Because it is in a different place:
```
# cat /sys/module/snd_sof/parameters/sof_debug
33281
```

**Comment by ujfalusi** [comment]
> @ujfalusi is this really necessary. A user can always read it from sysfs no? For example:
> 
> root@ranjani-Voxel:/sys/module/snd_sof_pci/parameters# cat sof_pci_debug 1024

My point was that it is not for the user, but it is for the developer on the other side of the world to have this information without waiting a day to get the value of the sof_debug for clarity.

I'm fine dropping this patch if it is seen as redundant

PS: I did not knew that our CI was using `sof_debug=0x41` ;)

**Comment by plbossart** [comment]
There's value in having this information in logs. Anything based on CI only provides us with the dmesg/journalctl logs, the module parameters are not all listed. They could but they aren't.

That said we could also add these kernel parameters to alsa-info, not sure if this is up-to-date and has the sof stuff.

**Comment by ranj063** [comment]
> There's value in having this information in logs. Anything based on CI only provides us with the dmesg/journalctl logs, the module parameters are not all listed. They could but they aren't.

reporting the moduel param in CI would be a trivial change @plbossart. It's not for the lack of information that we don't have this.

ANyway, please go ahead and merge this if everybody feels strongly about having this in dmesg.

**Pull request merged**
